src/SAWScript/Crucible/LLVM/MethodSpecIR.hs

```diff
@@ -196,6 +197,7 @@ data LLVMAllocSpec =
     , _allocSpecBytes :: Term
     , _allocSpecLoc   :: ProgramLoc
     , _allocSpecFresh :: Bool -- ^ Whether declared with @crucible_fresh_pointer@
+    , _allocSpecSymInit :: Bool
```

Comment:
Let's introduce a new datatype for the symInit argument instead of a boolean. That way we could easily add something like "zero this memory" as an allocation option too, and it makes the intent much more clear here and in the later use sites.